metals/src/main/scala/scala/meta/internal/metals/Embedded.scala

```diff
@@ -61,47 +62,60 @@ final class Embedded(
 }
 
 object Embedded {
-  def downloadSettings(
-      dependency: Dependency,
-      scalaVersion: String
-  ): Settings =
-    new coursiersmall.Settings()
-      .withTtl(Some(Duration.Inf))
-      .withDependencies(List(dependency))
-      .addRepositories(
-        List(
-          coursiersmall.Repository.SonatypeReleases,
-          coursiersmall.Repository.SonatypeSnapshots
+  lazy val repositories: mutable.Buffer[Repository] =
+    Repository.defaults().asScala ++
+      List(
+        Repository.central(),
+        Repository.ivy2Local(),
+        MavenRepository.of(
+          "https://oss.sonatype.org/content/repositories/releases/"
+        ),
+        MavenRepository.of(
+          "https://oss.sonatype.org/content/repositories/snapshots/"
         )
       )
-      .withForceVersions(
-        List(
-          new Dependency(
-            "org.scala-lang",
-            "scala-library",
-            scalaVersion
-          ),
-          new Dependency(
-            "org.scala-lang",
-            "scala-compiler",
-            scalaVersion
-          ),
-          new Dependency(
-            "org.scala-lang",
-            "scala-reflect",
-            scalaVersion
-          )
-        )
+
+  def dependency(dep: String): Dependency = {
+    val regex = """([^:]+):([^:]+):([^:]+)""".r
+
+    dep match {
+      case regex(org, name, version) =>
+        Dependency.of(org, name, version)
+    }
+  }
+
+  def dependency(dep: String*): Seq[Dependency] = dep.map(dependency)
+
+  def fetchSettings(
+      dep: Dependency,
+      scalaVersion: String
+  ): Fetch = {
+
+    val resolutionParams = ResolutionParams
+      .create()
+      .forceVersions(
+        dependency(
+          s"org.scala-lang:scala-library:$scalaVersion",
+          s"org.scala-lang:scala-compiler:$scalaVersion",
+          s"org.scala-lang:scala-reflect:$scalaVersion"
+        ).map(d => (d.getModule, d.getVersion)).toMap.asJava
       )
 
+    Fetch
+      .create()
+      .addRepositories(repositories: _*)
+      .withDependencies(dep)
+      .withResolutionParams(resolutionParams)
+      .withMainArtifacts()
+  }
+
   def newPresentationCompilerClassLoader(
       info: ScalaBuildTarget,
       scalac: ScalacOptionsItem
   ): URLClassLoader = {
-    val pc = new Dependency(
-      "org.scalameta",
-      s"mtags_${ScalaVersions.dropVendorSuffix(info.getScalaVersion)}",
-      BuildInfo.metalsVersion
+    val pc = dependency(
+      s"org.scalameta:mtags_${ScalaVersions
```

Comment:
extract the long expressions inside the string interpolation to a variable

Reply:
Changed.